src/main/java/com/powsybl/openloadflow/ac/outerloop/IncrementalShuntVoltageControlOuterLoop.java

```diff
@@ -156,20 +158,41 @@ public OuterLoopStatus check(AcOuterLoopContext context, Reporter reporter) {
         AcLoadFlowContext loadFlowContext = context.getLoadFlowContext();
         var contextData = (IncrementalContextData) context.getData();
 
-        List<LfShunt> controllerShunts = getControllerElements(contextData);
-        SensitivityContext sensitivityContext = new SensitivityContext(network, controllerShunts,
+        List<LfBus> controlledBuses = getControlledBuses(contextData);
+
+        // check which shunts are not within their deadbands
+        List<LfShunt> controllerShuntsOutsideOfDeadband = new ArrayList<>();
+        List<LfBus> controlledBusesOutsideOfDeadband = new ArrayList<>();
+        controlledBuses.forEach(controlledBus -> {
+            ShuntVoltageControl voltageControl = controlledBus.getShuntVoltageControl().orElseThrow();
+            double diffV = controlledBus.getHighestPriorityMainVoltageControl().orElseThrow().getTargetValue() - voltageControl.getControlledBus().getV();
+            double halfTargetDeadband = getHalfTargetDeadband(voltageControl);
+            if (Math.abs(diffV) > halfTargetDeadband) {
+                List<LfShunt> controllers = voltageControl.getMergedControllerElements().stream()
```

Comment:
We have a difference here in the way we get controllers from a controlled. In previous code, we use `IncrementalContextData.getControllerElements` which has a filter on voltage control type. Is it equivalent?

Reply:
Yes, it is equivalent. I directly call the method for the specific type of voltage control: `getShuntVoltageControl` or `getTransformerVoltageControl`. And I also filter out the disabled controllers.